src/backends/pgfplotsx.jl

```diff
@@ -0,0 +1,813 @@
+using Contour: Contour
+Base.@kwdef mutable struct PGFPlotsXPlot
+    is_created::Bool = false
+    was_shown::Bool = false
+    the_plot::PGFPlotsX.TikzDocument = PGFPlotsX.TikzDocument()
+    function PGFPlotsXPlot(is_created, was_shown, the_plot)
+        pgfx_plot = new(is_created, was_shown, the_plot)
+        # tikz libraries
+        PGFPlotsX.push_preamble!(pgfx_plot.the_plot, "\\usetikzlibrary{arrows.meta}")
+        PGFPlotsX.push_preamble!(pgfx_plot.the_plot, "\\usetikzlibrary{backgrounds}")
+        PGFPlotsX.push_preamble!(pgfx_plot.the_plot,
+        """
+        \\pgfkeys{/tikz/.cd,
+          background color/.initial=white,
+          background color/.get=\\backcol,
+          background color/.store in=\\backcol,
+        }
+        \\tikzset{background rectangle/.style={
+            fill=\\backcol,
+          },
+          use background/.style={
+            show background rectangle
+          }
+        }
+        """
+        )
+        # pgfplots libraries
+        PGFPlotsX.push_preamble!(pgfx_plot.the_plot, "\\usepgfplotslibrary{patchplots}")
+        PGFPlotsX.push_preamble!(pgfx_plot.the_plot, "\\usepgfplotslibrary{fillbetween}")
+        pgfx_plot
+    end
+end
+
+function pgfx_axes(pgfx_plot::PGFPlotsXPlot)
+    gp =  pgfx_plot.the_plot.elements[1].elements[1]
+    return gp isa PGFPlotsX.GroupPlot ? gp.contents : gp
+end
+
+function pgfx_preample(pgfx_plot::Plots.Plot{Plots.PGFPlotsXBackend})
```

Comment:
Looks like a typo

Reply:
what exactly?

Reply:
Preample?

Reply:
:D, you are right, didn't saw that

Reply:
Is it used anywhere?

Reply:
its a utility function for someone (like me) who is embedding the tex output in documents and needs to know what custom preamble is to be included.

Reply:
Alright, might make sense to add a comment or something so someone doesn't just go and delete it :;P